src/shared/util.js

```diff
@@ -1219,16 +1219,111 @@ function MessageHandler(sourceName, targetName, comObj) {
   this.targetName = targetName;
   this.comObj = comObj;
   this.callbackIndex = 1;
+  this.streamId = 1;
   this.postMessageTransfers = true;
-  var callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
-  var ah = this.actionHandler = Object.create(null);
+  let streamSinks = this.streamSinks = Object.create(null);
+  let streamControllers = this.streamControllers = Object.create(null);
+  let callbacksCapabilities = this.callbacksCapabilities = Object.create(null);
+  let ah = this.actionHandler = Object.create(null);
 
   this._onComObjOnMessage = (event) => {
-    var data = event.data;
+    let data = event.data;
     if (data.targetName !== this.sourceName) {
       return;
     }
-    if (data.isReply) {
+    if (data.stream) {
+      let sourceName = this.sourceName;
+      let targetName = data.sourceName;
+      let streamId = data.streamId;
+      switch (data.stream) {
+        case 'start_complete':
+          if (data.success === true || data.success === 'true') {
```

Comment:
> That shouldn't be a problem, since the structured cloning algorithm maintains types (and throws on unsupported types).

Okay right, we are using _structured cloning algorithm_ in `postMessage`.
Thanks for clarification.